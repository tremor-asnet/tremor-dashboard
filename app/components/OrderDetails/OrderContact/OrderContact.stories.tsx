import type { Meta, StoryObj } from "@storybook/react";

// Components
import OrderContact from "./OrderContact";

const propsDefault = {
  name: "Leather Wallet",
  url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
  date: "2024-01-11T09:48:52Z",
};

const meta = {
  title: "Components/OrderContact",
  component: OrderContact,
  tags: ["autodocs"],
} as Meta<typeof OrderContact>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  render: () => <OrderContact {...propsDefault} />,
};
