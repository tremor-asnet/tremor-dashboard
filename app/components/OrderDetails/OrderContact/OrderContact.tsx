// Components
import { Button, Flex, Title, Text } from "@tremor/react";
import { CustomImage } from "@/components";

const OrderContact = ({ name, url }: { name: string; url: string }) => {
  // TODO: Handle delivered days
  return (
    <Flex>
      <Flex justifyContent="start">
        <div className="mr-4">
          <CustomImage
            alt={name}
            className="rounded-full"
            height={110}
            priority
            src={url}
            width={110}
          />
        </div>
        <Flex flexDirection="col" alignItems="start" className="py-4">
          <Title className="text-tremor-content-title dark:text-dark-tremor-content-title text-primary font-semibold capitalize leading-[26px] tracking-[0.12px] truncate max-w-[100px] lg:max-w-[200px] xl:max-w-[300px] 2xl:max-w-[400px] min-w-[100px]">
            {name}
          </Title>
          <p className="text-sm dark:text-dark-romance font-light opacity-100 text-secondary leading-[21px] tracking-[0.4px] truncate max-w-[100px] lg:max-w-[200px] xl:max-w-[300px] 2xl:max-w-[400px] min-w-[100px]">
            Order was delivered 2 days ago.
          </p>
          <p className="p-2 mt-4 font-bold text-white dark:text-white bg-green-500 text-xs rounded-tremor-small leading-[10.5px] tracking-[0.18px] uppercase">
            delivered
          </p>
        </Flex>
      </Flex>
      <Flex>
        <Flex className="flex-col items-end">
          <Button className="antialiased min-w-[64px] py-[12px] text-center uppercase sm:px-4 bg-gradient-primary dark:bg-gradient-pickled rounded-lg border-0 hover:shadow-btn-primary-hover px-2 py-1.5 leading-[17px] tracking-[0.35px]">
            <Text className="uppercase py-[2px] text-xs font-bold text-white uppercase">
              contact us
            </Text>
          </Button>
          <Text className="mt-3 text-sm dark:text-dark-romance font-light opacity-100 text-secondar leading-[21px] tracking-[0.4px] truncate max-w-[100px] lg:max-w-[200px] xl:max-w-[300px] 2xl:max-w-[400px] min-w-[100px]">
            Do you like the item? Leave us a review here.
          </Text>
        </Flex>
      </Flex>
    </Flex>
  );
};

export default OrderContact;
