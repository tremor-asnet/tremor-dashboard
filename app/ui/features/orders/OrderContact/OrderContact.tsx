import Link from "next/link";

// Components
import { Flex, Title, Text } from "@tremor/react";
import { OrderStatus } from "./OrderStatus";
import { Button, CustomImage } from "@/ui/components";

// Constants
import { VARIANT_BUTTON } from "@/constants";

const OrderContact = ({
  name,
  url,
  date,
  status = 0,
}: {
  name: string;
  url: string;
  date: number;
  status?: number;
}) => {
  return (
    <Flex className="flex-wrap sm:flex-nowrap order-contact">
      <Flex justifyContent="start">
        <div className="mr-4">
          <CustomImage
            alt={name}
            className="rounded-full"
            height={110}
            priority
            src={url}
            width={110}
          />
        </div>
        <Flex flexDirection="col" alignItems="start" className="py-4">
          <Title className="text-tremor-content-title dark:text-dark-tremor-content-title text-primary font-semibold capitalize leading-6 tracking-wide truncate max-w-[200px] md:max-w-[240px] xl:max-w-xs 2xl:max-w-sm">
            {name}
          </Title>
          <OrderStatus status={status} period={date} />
        </Flex>
      </Flex>
      <Flex>
        <Flex flexDirection="col" alignItems="end">
          <Button
            variant={VARIANT_BUTTON.PRIMARY}
            additionalClass="antialiased min-w-[64px] py-[12px] text-center uppercase sm:px-4 rounded-lg border-0 px-2 py-1.5 leading-[17px] tracking-[0.35px]">
            <Text className="uppercase py-[2px] text-xs font-bold text-white dark:text-dark-primary uppercase">
              contact us
            </Text>
          </Button>
          <Text className="mt-3 text-sm dark:text-dark-romance font-light opacity-100 text-secondary text-right leading-[21px] w-full">
            Do you like the product? Leave us a review{" "}
            <Link href="#" className="font-normal">
              here
            </Link>
            .
          </Text>
        </Flex>
      </Flex>
    </Flex>
  );
};

export default OrderContact;
