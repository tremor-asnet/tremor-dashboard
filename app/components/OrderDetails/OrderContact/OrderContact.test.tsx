import { render } from "@testing-library/react";

// Components
import OrderContact from "./OrderContact";

describe("Testing Order detail contact section", () => {
  const propsDefault = {
    name: "Leather Wallet",
    url: "https://demos.creative-tim.com/nextjs-material-dashboard-pro//_next/static/media/black-chair.b2719b4f.jpeg",
  };

  it("Should match snapshot", () => {
    const component = render(<OrderContact {...propsDefault} />);
    expect(component).toMatchSnapshot();
  });
});
