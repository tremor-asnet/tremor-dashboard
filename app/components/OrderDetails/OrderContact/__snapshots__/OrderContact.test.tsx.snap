// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Order detail contact section Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center flex-wrap sm:flex-nowrap order-contact"
      >
        <div
          class="tremor-Flex-root flex w-full flex-row justify-start items-center"
        >
          <div
            class="mr-4"
          >
            <img
              alt="Leather Wallet"
              class="rounded-full"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="110"
              src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=256&q=75"
              srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=256&q=75 2x"
              style="color: transparent;"
              width="110"
            />
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-start py-4"
          >
            <p
              class="text-tremor-title dark:text-dark-tremor-content-title text-primary font-semibold capitalize leading-[26px] tracking-[0.12px] truncate max-w-[100px] lg:max-w-[200px] xl:max-w-[300px] 2xl:max-w-[400px] min-w-[100px]"
            >
              Leather Wallet
            </p>
            <p
              class="text-sm dark:text-dark-romance font-light opacity-100 text-secondary leading-[21px] w-full"
            >
              Order was delivered 
              18
               days ago.
            </p>
            <p
              class="p-2 mt-4 font-bold text-white dark:text-white bg-green-500 text-xs rounded-tremor-small leading-[10.5px] tracking-[0.18px] uppercase"
            >
              delivered
            </p>
          </div>
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
        >
          <div
            class="tremor-Flex-root flex w-full justify-between flex-col items-end"
          >
            <button
              class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis antialiased min-w-[64px] text-center uppercase sm:px-4 bg-gradient-primary dark:bg-gradient-pickled rounded-lg border-0 px-2 py-1.5 leading-[17px] tracking-[0.35px]"
            >
              <span
                class="tremor-Button-text text-sm whitespace-nowrap"
              >
                <p
                  class="text-tremor-default py-[2px] text-xs font-bold text-white dark:text-dark-primary uppercase"
                >
                  contact us
                </p>
              </span>
            </button>
            <p
              class="text-tremor-default mt-3 text-sm dark:text-dark-romance font-light opacity-100 text-secondary text-right leading-[21px] w-full"
            >
              Do you like the product? Leave us a review
               
              <a
                class="font-normal"
                href="#"
              >
                here
              </a>
              .
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tremor-Flex-root flex w-full flex-row justify-between items-center flex-wrap sm:flex-nowrap order-contact"
    >
      <div
        class="tremor-Flex-root flex w-full flex-row justify-start items-center"
      >
        <div
          class="mr-4"
        >
          <img
            alt="Leather Wallet"
            class="rounded-full"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="110"
            src="http://localhost/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fdemos.creative-tim.com%2Fnextjs-material-dashboard-pro%2F%2F_next%2Fstatic%2Fmedia%2Fblack-chair.b2719b4f.jpeg&w=256&q=75 2x"
            style="color: transparent;"
            width="110"
          />
        </div>
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-start py-4"
        >
          <p
            class="text-tremor-title dark:text-dark-tremor-content-title text-primary font-semibold capitalize leading-[26px] tracking-[0.12px] truncate max-w-[100px] lg:max-w-[200px] xl:max-w-[300px] 2xl:max-w-[400px] min-w-[100px]"
          >
            Leather Wallet
          </p>
          <p
            class="text-sm dark:text-dark-romance font-light opacity-100 text-secondary leading-[21px] w-full"
          >
            Order was delivered 
            18
             days ago.
          </p>
          <p
            class="p-2 mt-4 font-bold text-white dark:text-white bg-green-500 text-xs rounded-tremor-small leading-[10.5px] tracking-[0.18px] uppercase"
          >
            delivered
          </p>
        </div>
      </div>
      <div
        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
      >
        <div
          class="tremor-Flex-root flex w-full justify-between flex-col items-end"
        >
          <button
            class="tremor-Button-root flex-shrink-0 inline-flex justify-center items-center group font-medium outline-none rounded-tremor-default shadow-tremor-input dark:shadow-dark-tremor-input text-sm border-tremor-brand dark:border-dark-tremor-brand text-tremor-brand-inverted dark:text-dark-tremor-brand-inverted hover:bg-tremor-brand-emphasis dark:hover:bg-dark-tremor-brand-emphasis hover:border-tremor-brand-emphasis dark:hover:border-dark-tremor-brand-emphasis antialiased min-w-[64px] text-center uppercase sm:px-4 bg-gradient-primary dark:bg-gradient-pickled rounded-lg border-0 px-2 py-1.5 leading-[17px] tracking-[0.35px]"
          >
            <span
              class="tremor-Button-text text-sm whitespace-nowrap"
            >
              <p
                class="text-tremor-default py-[2px] text-xs font-bold text-white dark:text-dark-primary uppercase"
              >
                contact us
              </p>
            </span>
          </button>
          <p
            class="text-tremor-default mt-3 text-sm dark:text-dark-romance font-light opacity-100 text-secondary text-right leading-[21px] w-full"
          >
            Do you like the product? Leave us a review
             
            <a
              class="font-normal"
              href="#"
            >
              here
            </a>
            .
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
